### PR TITLE
Enhance etcd documentation

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR enhances the documentation about etcd and ETCD Druid usage in Gardener.

**Which issue(s) this PR fixes**:
Fixes a part of #3405

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/usage/etcd.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# etcd - Key-Value Store for Kubernetes
+
+[etcd](https://etcd.io/) is a strongly consistent key-value store and the most prevalent choice for the Kubernetes
+persistence layer. All API cluster objects like `Pod`s, `Deployment`s, `Secret`s, etc. are stored in `etcd` which
+makes it an essential part of a [Kubernetes control plane](https://kubernetes.io/docs/concepts/overview/components/#control-plane-components).
+
+## Shoot cluster persistence
+
+Each shoot cluster gets its very own persistence for the control plane. It runs in the shoot namespace on the respective
+seed cluster and is managed by a `StatefulSet`. Concretely, there are two `StatefulSets`s (with `volumeClaimTemplates`)
```
Comment (vpnachev) [inline docs/usage/etcd.md]: Looking into https://github.com/gardener/gardener/blob/master/docs/proposals/06-etcd-druid.md, the statefulsets should be abstracted via the druid and the ETCD resource. So, I think from gardener/gardener point of view this documentation must be limited to the ETCD resources and secrets, but implementation details like StatefulSets, volumeClaimTemplates, etc. should be documented in https://github.com/gardener/etcd-druid/.

Comment (timuthy) [reply docs/usage/etcd.md]: Would it be okay for you if I write this sentence as an implementation **example**? You are right about the abstraction. My goal was also to give the reader an idea about how this is concretely implemented w/o pointing to another documentation. The Druid dependency is hard-coded and well known, so I would say it doesn't hurt.

Comment (vpnachev) [reply docs/usage/etcd.md]: As far as there is no plans to change the implementation behind the ETCD (for example, change the statefulsets to  deployments), I think it is fine.

File: docs/usage/etcd.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# etcd - Key-Value Store for Kubernetes
```
Comment (rfranzke) [inline docs/usage/etcd.md]: Should this file be moved into the `concepts` directory?

File: docs/usage/etcd.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Gardenlet deletes the `Etcd` resource and [ETCD Druid](https://github.com/gardener/etcd-druid/) takes care about cleaning up
+all related objects, e.g. the backing `StatefulSet`.
+
+## Autoscaling
+
+Gardenlet maintains [HVPA](https://github.com/gardener/hvpa-controller/blob/master/config/samples/autoscaling_v1alpha1_hvpa.yaml)
+objects for etcd `StatefulSet`s if the corresponding [feature gate](../deployment/feature_gates.md) is enabled. This enables
+a vertical scaling for `etcd`. Downscaling is handled more pessimistic to prevent many subsequent `etcd` restarts. Thus,
+for `production` clusters downscaling is deactivated and for all other clusters lower advertised requests/limits are only
+considered during a shoot's maintenance time window.
```
Comment (rfranzke) [inline docs/usage/etcd.md]:
```suggestion
applied during a shoot's maintenance time window.
```

File: docs/usage/etcd.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+page https://github.com/gardener/etcd-backup-restore.
+
+How long backups are stored in the bucket after a shoot has been deleted, depends on the configured _retention period_ in the
+`Seed` resource. Please see this [example configuration](https://github.com/gardener/gardener/blob/849cd857d0d20e5dde26b9740ca2814603a56dfd/example/20-componentconfig-gardenlet.yaml#L20) for more information.
+
+## Housekeeping
+
+[etcd maintenance tasks](https://etcd.io/docs/v3.3/op-guide/maintenance/) must be performed from time to time in order
+to re-gain database storage and to ensure the system's reliability. The [backup-restore](https://github.com/gardener/etcd-backup-restore)
+_sidecar_ takes care about this job, as well. Gardener chooses a random time **within the shoot's maintenance time** to
```
Comment (rfranzke) [inline docs/usage/etcd.md]:
```suggestion
_sidecar_ takes care about this job as well. Gardener chooses a random time **within the shoot's maintenance time** to
```

File: docs/usage/etcd.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+configure these tasks.
```
Comment (rfranzke) [inline docs/usage/etcd.md]:
```suggestion
schedule these tasks.
```

**Comment by timuthy** [comment]
Thanks for your reviews @rfranzke and @vpnachev. PTAL.
/merge squash

**Review by vpnachev — APPROVED** [review]
/lgtm
/milestone v1.32

**Review by rfranzke — APPROVED** [review]

**Pull request merged**